src/java.base/share/classes/java/util/zip/InflaterInputStream.java

```diff
@@ -127,10 +127,15 @@ public int read() throws IOException {
      * Reads uncompressed data into an array of bytes. If {@code len} is not
      * zero, the method will block until some input can be decompressed; otherwise,
      * no bytes are read and {@code 0} is returned.
+     *
+     * Unlike the {@link InputStream#read(byte[],int,int) overridden method}
+     * of {@code InputStream}, this method might write more bytes than the returned
+     * number of inflated bytes into the buffer {@code b}.
```

Comment:
Hello Volker, I think this comment should be a bit more clear and state what exactly it means by writing more bytes than the returned value. Specifically, I think it should clearly state that even though it might write additional data, it will however not write/change data starting and beyond `off + len` index in the passed array.

Reply:
I think this require a bit of word smithing. If the return value is 'n' then it should make it clear that the values in elements b[off + n] to b[off + len - 1] are undefined, their values may or may have been changed by the inflate/read operation. For completeness, the expectation for when inflate fails should be specified too, the contents of b[off] to b[off + len - 1] will be undefined.

Reply:
Thanks for your comments. I've updated the API-doc according to your recommendations. Please feel free to propose new/better wording :)

I'll wait with updating the CSR until we reach final agreement in this PR.